### ISSUE TITLE
Add S3 object store storage class option

### ISSUE THREAD

**Issue by aentwist** [nextcloud/docker#1932]
## User Story

When using S3-compatible object stores, choosing the storage class for objects is an important consideration. By tailoring the storage class to the application requirements there can be massive cost savings.

As a small-time user I want to be able to choose a cost-efficient storage class for light yet bursty usage (intelligent tiering). I do not want to mount additional files / add custom configurations to the base Docker image. I want to interact with this setting the same as with all other similar settings - by environment variable.

## Proposed solution

Recently, this feature appears to have been implemented into nextcloud/server via https://github.com/nextcloud/server/pull/36075. The goal of this issue is to add it to the swath of environment variable settings available for the Docker image as described in the README.

<blockquote>
To use an external S3 compatible object store as primary storage, set the following variables:

    OBJECTSTORE_S3_HOST: The hostname of the object storage server
    OBJECTSTORE_S3_BUCKET: The name of the bucket that Nextcloud should store the data in
    OBJECTSTORE_S3_KEY: AWS style access key
    OBJECTSTORE_S3_SECRET: AWS style secret access key
    OBJECTSTORE_S3_PORT: The port that the object storage server is being served over
    OBJECTSTORE_S3_SSL (default: true): Whether or not SSL/TLS should be used to communicate with object storage server
    OBJECTSTORE_S3_REGION: The region that the S3 bucket resides in.
    OBJECTSTORE_S3_USEPATH_STYLE (default: false): Not required for AWS S3
    OBJECTSTORE_S3_LEGACYAUTH (default: false): Not required for AWS S3
    OBJECTSTORE_S3_OBJECT_PREFIX (default: urn:oid:): Prefix to prepend to the fileid
    OBJECTSTORE_S3_AUTOCREATE (default: true): Create the container if it does not exist

Check the [Nextcloud documentation](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/primary_storage.html#simple-storage-service-s3) for more information.
</blockquote>

Following the naming convention and [current default storage class of STANDARD](https://github.com/nextcloud/server/pull/36075/files#diff-32333c316cf51e92bd3b53808f0fe1eca1635fb1f62daba1469ce6694959f97cR87), let's go with

```
OBJECTSTORE_S3_STORAGE_CLASS (default: STANDARD): The storage class to apply to all stored objects
```

## Alternative Considerations

- How will this behave for S3 compatible stores not S3? Do they all support it? In any case, that is probably out of scope of this issue. We just need to interact with the setting that has already been implemented.

## Notes

- available from Nextcloud 26
- currently undocumented

**Comment by aentwist** [comment]
I *really* want this. Probably going to pull for it immediately..

**Comment by aentwist** [comment]
Going to look at adding it to 25. If you want it otherwise, then backport it.

**Comment by aentwist** [comment]
Looks like from the git blame the versions are automated... as strange as this setup is. Pretty simple update then.

**Comment by aentwist** [comment]
Exclusive to Nextcloud 26, needs noted in the README

**Comment by aentwist** [comment]
workaround: get the updated s3 config file from the MR and mount it over the existing s3 config file at /usr/src/nextcloud/config/s3.config.php to support `OBJECTSTORE_S3_STORAGE_CLASS`. unfortunately nextcloud does not deep merge configs, so if you don't mount over it then it will be overridden by the existing s3.config.php, so you have to provide a replacement for the entire file.

**Comment by joshtrichards** [comment]
Fixed in #2138

**Comment by joshtrichards** [comment]
@Benjamin-Watson0 Then you don't need to use this parameter.
